mono/metadata/threads.c

```diff
@@ -270,18 +270,23 @@ mono_threads_begin_abort_protected_block (void)
 		new_state |= new_val << ABORT_PROT_BLOCK_SHIFT;
 
 	} while (InterlockedCompareExchangePointer ((volatile gpointer)&thread->thread_state, (gpointer)new_state, (gpointer)old_state) != (gpointer)old_state);
+
+	/* We defer also the global thread interruption count */
+	if (old_state & INTERRUPT_REQUESTED_BIT)
```

Comment:
I'm not sure this changes play well with how thread_interruption_requested is bumped by callers of  mono_thread_clear_interruption_requested and mono_thread_clear_interruption_requested.

They might be decrementing when there's a deferred interrupt that didn't bump in the first place.